internal/native/native.go

```diff
@@ -61,61 +56,49 @@ func (s VideoStreamingStatus) String() string {
 }
 
 func NewNative(opts NativeOptions) *Native {
-	pid := os.Getpid()
-	nativeSubLogger := nativeLogger.With().Int("pid", pid).Str("scope", "native").Logger()
-	displaySubLogger := displayLogger.With().Int("pid", pid).Str("scope", "native").Logger()
-
-	onVideoStateChange := opts.OnVideoStateChange
-	if onVideoStateChange == nil {
-		onVideoStateChange = func(state VideoState) {
-			nativeLogger.Info().Interface("state", state).Msg("video state changed")
-		}
+	n := &Native{
+		ready:                make(chan struct{}),
+		systemVersion:        opts.SystemVersion,
+		appVersion:           opts.AppVersion,
+		displayRotation:      opts.DisplayRotation,
+		defaultQualityFactor: opts.DefaultQualityFactor,
+		onVideoStateChange:   opts.OnVideoStateChange,
+		onVideoFrameReceived: opts.OnVideoFrameReceived,
+		onIndevEvent:         opts.OnIndevEvent,
+		onRpcEvent:           opts.OnRpcEvent,
+		sleepModeSupported:   isSleepModeSupported(),
+		videoLock:            sync.Mutex{},
 	}
 
-	onVideoFrameReceived := opts.OnVideoFrameReceived
-	if onVideoFrameReceived == nil {
-		onVideoFrameReceived = func(frame []byte, duration time.Duration) {
-			nativeLogger.Trace().Interface("frame", frame).Dur("duration", duration).Msg("video frame received")
+	if n.onVideoStateChange == nil {
+		n.onVideoStateChange = func(state VideoState) {
+			GetDisplayLogger().Info().Interface("state", state).Msg("video state changed")
```

Comment:
Here's an example of callbacks needing to ask for the logger at callback-time instead of whatever it was when this handler was setup.